pages/verify/[discordServerId]/[discordMemberId]/[customLink].tsx

```diff
@@ -54,6 +54,42 @@ const VerifyPage = ({
   const [verifiedSignature, setVerifiedSignature] = useState(false);
   const [unverifiedSignature, setUnverifiedSignature] = useState("");
   const [chainId, setChainId] = useState("");
+  const [isArgent, setIsArgent] = useState(false);
+  const [isSwitching, setIsSwitching] = useState(false);
+
+  const getTargetChainId = () => {
+    return chainAliasByNetwork[starknetNetwork][1];
+  };
+
+  const handleNetworkSwitch = async (wallet: any) => {
```

Comment:
You have an example of useCallback at line 94. It's very similar, but it avoid rerenders with react!